lib/semantics/tools.h

```diff
@@ -113,10 +113,49 @@ inline bool IsAssumedSizeArray(const Symbol &symbol) {
 }
 bool IsAssumedLengthCharacter(const Symbol &);
 bool IsAssumedLengthCharacterFunction(const Symbol &);
+// Is the symbol modifiable in this scope
 std::optional<parser::MessageFixedText> WhyNotModifiable(
     const Symbol &symbol, const Scope &scope);
-// Is the symbol modifiable in this scope
 bool IsExternalInPureContext(const Symbol &symbol, const Scope &scope);
+bool HasCoarray(const parser::Expr &expression);
+template<typename T>
+
+// Detection of image control statements.  If the parameter is not an image
```

Comment:
This is way more code than necessary, and it conflates two separate problems: determining whether a given parse tree node type represents an image control statement, and mapping those node types to error messages (or strings to be used in a common error message).

Reply:
Thanks, Peter and Jean.

With respect to reducing the amount of code -- I note that all of these statements are direct or indirect members of the variant ```parser::ExecutableConstruct```.  So my plan is to build the interface simply around that type.

I also plan to change the interface to not return a string and to simply return a boolean.  Specifically, here's the interface I plan to implement --
```bool IsImageControlStatement(parser::ExecutableConstruct);```

Let me know if there's a better way.